### ISSUE TITLE
(Feature Request) Smb file storage 

### ISSUE THREAD

**Issue by RubyEyedReaper** [XITRIX/iTorrent#352]
Being able to change the download location to an SMB rather than just local file storage, this would make all the difference

**Comment by XITRIX** [comment]
You already can do that
You can add SMB server into Files app, than open `iTorrent -> Preferences -> Storage` and add this SMB server as storage for iTorrent
LibTorrent not really optimized to have slow read\write access and could lag because of that, but at least it should work

**Comment by CrazyFriedRice** [comment]
I'm assuming this is some sort of user error here, but any idea if this is normal?![image](https://github.com/user-attachments/assets/66d3b2b8-24bf-4c81-809d-9e5e8acc8604)

![image](https://github.com/user-attachments/assets/a153af2a-e446-430f-b6d8-620f1a71dfbe)

Shows up as a read only drive for me. Works fine in the Files app and also VLC app. Able to read/write. I approved all permissions. Advice?

**Comment by CrazyFriedRice** [comment]
Disregard. I'm an idiot. Logged in as guest on smb. 🤦‍♂️

**Comment by CrazyFriedRice** [comment]
Actually it now shows read only in Files app and other apps as well. Do I need to be jailbroken perhaps?

**Comment by XITRIX** [comment]
No, it's SMB server setting, you need to allow specific filesystem in config. Honestly, I forgot it's name, but I found it by Googling something like "iOS Files SMB server read only" 

UP. 
Wait, you said that you can write in Files app, but in iTorrent it's read only? If yes, it's weird, it should have the same access as regular Files app

**Comment by CrazyFriedRice** [comment]
Absolutely correct! Interesting issue this one. I managed to regain some write control using the owlfiles app to modify, rename, etc before moving to smb. Issue isn't with the app. 🙌

**Comment by XITRIX** [comment]
Btw. on screenshot you are in root of host, which is read only for sure. You need to select a folder where iTorrent will write it's files

**Comment by CrazyFriedRice** [comment]
After doing a bit more googling, I see what you meant by host root now. Thanks for explaining! Very much appreciated:)

**Comment by MakiCz** [comment]
Hi, I’ve added new folder in smb server (local synology) and chose  it in iTorrent, but it is still saving into default folder. I have to manually cut and paste downloaded files from iTorrent folder via “Files”. Reading and writing work in Files. Any solution please?

Edit: I tried to create one new folder in internal iPad storage, but it isn’t working either.
